### ISSUE TITLE
AL Loop for Weights

### ISSUE THREAD

**Issue by tpr0p** [slmsuite/slmsuite#59]
Add the [Augmented Lagrangian method](https://en.wikipedia.org/wiki/Augmented_Lagrangian_method) for increasing weights in experimental optimization.

**Comment by ichristen** [comment]
I agree that it would be interesting to try this. I don’t know of any work that looks into this. People use very simple weighting functions. In the 3d_spots branch, we have a tanh test function that I think will be more stable. I suspect that people don’t use AL because balancing the weights is a very nonlinear function and it’s better to just go very slowly and use something that always works instead of tuning AL. But would be cool to try!
